src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -623,19 +635,16 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, pal::stri
             }
         }
     }
-    else
-    {
-        trace::verbose(_X("Failed to obtain current working dir"));
-    }
 
     std::vector<pal::string_t> hive_dir;
     pal::string_t local_dir;
     pal::string_t global_dir;
-    bool multilevel_lookup = multilevel_lookup_enabled();
+
+    bool multilevel_lookup = own_dir.empty() ? true : multilevel_lookup_enabled();
```

Comment:
Your change above says "Always do multilevel lookup when own_dir is empty". If the user disabled multi-level lookup (to be semantically equivalent to 1.X behavior) and own_dir was empty, you will still probe the other multi-level lookup locations, violating what the user wanted. This is incorrect.

Multi-level lookup is orthogonal to whether the location is specified or not.